pkgs/top-level/aliases.nix

```diff
@@ -943,10 +943,7 @@ mapAliases ({
   nixFlakes = nixVersions.stable; # Added 2021-05-21
   nixStable = nixVersions.stable; # Added 2022-01-24
   nixUnstable = nixVersions.unstable; # Added 2022-01-26
-  nix_2_3 = nixVersions.nix_2_3;
-  nix_2_4 = nixVersions.nix_2_4;
-  nix_2_5 = nixVersions.nix_2_5;
-  nix_2_6 = nixVersions.nix_2_6;
+  inherit (nixVersions) nix_2_3 nix_2_4 nix_2_5 nix_2_6 nix_2_7 nix_2_8 nix_2_9 nix_2_10;
```

Comment:
Shouldn't we remove the older alias them? I found this because I couldn't auto-complete `nix-build -A nix_2_` and them got a useless error when I forced type `nix-build -A nix_2_10` as:

```
error: attribute 'nix_2_10' in selection path 'nix_2_10' not found
       Did you mean one of nix_2_3, nix_2_4, nix_2_5 or nix_2_6?
```

And I was fairly sure that my nixpkgs was out-of-date or something until I realized that we have `nixVersions` attrset. So I would argue that just keeping a feel of the alias at the top-level is just hurting discovery for no reason.

Reply:
Removed the commit, but my point above still is valid.

Reply:
@lovesegfault I pushed one more commit that throws with a better error message if you try to use the nix versioned top-level aliases, with the exception of `nix_2_3` (that may still being used around).